src/lib/stream/chacha/chacha.h

```diff
@@ -42,13 +42,15 @@ class BOTAN_DLL ChaCha final : public StreamCipher
 
       std::string name() const override;
 
+      void seek(u64bit offset) override;
+
    private:
       void key_schedule(const byte key[], size_t key_len) override;
 
       size_t m_rounds;
       secure_vector<u32bit> m_state;
       secure_vector<byte> m_buffer;
-      size_t m_position = 0;
+      u64bit m_position = 0;
```

Comment:
Why this change? m_position is just an index into the (small) vector `m_buffer`, and could even (besides consistency concerns) just be a short rather than a size_t.